evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +122,52 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectDtypeColumns(ColumnSelector):
+    """
+    Selects columns by specified datatype in input data.
+
+    Arguments:
+        columns (list(string), list(ww.LogicalType)): List of datatypes, used to determine which columns to select.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Select Datatype Columns Transformer"
+    hyperparameter_ranges = {}
+    needs_fitting = False
+
+    def _check_input_for_columns(self, X):
+        cols = self.parameters.get("columns") or []
+
+        if len(cols) > 0 and not isinstance(cols[0], str):
+            column_types = [
+                logical_type.__class__ for logical_type in X.ww.logical_types.values()
+            ]
+        else:
+            column_types = [
+                logical_type.type_string for logical_type in X.ww.logical_types.values()
+            ]
```

Comment:
This logic is definitely not the cleanest it could be - if anyone knows a more efficient way to pull out this information, please let me know!